src/ifcopenshell-python/ifcopenshell/util/shape.py

```diff
@@ -730,3 +730,16 @@ def get_extrusions(element: ifcopenshell.entity_instance) -> list[ifcopenshell.e
             else:
                 break
     return extrusions
+
+
+def get_total_edge_length(geometry: ShapeType) -> float:
+    """Calculates the total length of edges in a given geometry.
+
+    :param geometry: Geometry output calculated by IfcOpenShell
+    :type geometry: geometry
+    :return: The total length of all edges in the geometry.
+    :rtype: float
+    """
+    vertices = get_vertices(geometry)
+    edges = get_edges(geometry)
+    return sum([np.linalg.norm(vertices[e[0]] - vertices[e[1]]) for e in edges])
```

Comment:
You can probably resolve this all in numpy for some insignificant speedup. Untested:

```suggestion
    edges = vertices[get_edges(geometry)]
    return np.linalg.norm(vertices[:, 1] - vertices[:, 0], axis=1).sum()
```

Reply:
Did you mean this?:
`vertices = vertices[get_edges(geometry)]`
instead of this?:
`edges = vertices[get_edges(geometry)]`

Reply:
Ah yes that's what I meant. Thanks